DEP 12: add support for the Rectangle Rule

cgay opened the pull request.

waywardmonkeys reviewed (commented):
Does the whitespace before the lines of text and before the close delimiter need to match exactly in terms of the characters used? I think better to specify that the sequence of characters in the whitespace must match exactly. (Since you can't just outright ban tabs or assume a given width for the tab...)

cgay commented:
> Does the whitespace before the lines of text and before the close delimiter need to match exactly in terms of the characters used? I think better to specify that the sequence of characters in the whitespace must match exactly. (Since you can't just outright ban tabs or assume a given width for the tab...)

Great point, thanks!  I added a bullet point to the multi-line rules section.

Unrelatedly, I removed the bullet point 

* Multi-line raw string literals require both opening and closing quote characters on their own line.

because it's not necessary for the initial `"""` to be on a line by itself. (The C# examples don't put it on a line by itself, so their docs are inconsistent.)
